### PR TITLE
Fix #106, explain how to deal with undeterminable and anon authors

### LINKED ISSUE #106: Document how to deal with anonymous authors
Somewhat related to #103.

We should discuss whether to introduce an enum to reflect non-retrievable and/or unknown authors, e.g. `UNKNOWN`, for when CFF files are auto-generated. This way, we can avoid introducing obfuscating pseudo-identities, which become interesting only when citation information is presented to human users of the software.

### PR DESCRIPTION
**Pull request by sdruskat**
- Explains how to deal with undeterminable and anon. authors
- References the new section from the "referencing other works" information
section

**Related issues**

Refs: #240 (tracking implementation), fix #106 (describing issue)

**Describe the changes made in this pull request**

- Adds an explanatory subsection to `definition.reference.authors` section
- Add links to new section from all authors sections.

**Instructions to review the pull request**
- Look at `schema-guide.md`
- Discuss and decide whether
    1. this is the best place to describe this?
    2. the header (needed to link to it) is the right level
    3. references to new section are in all (and only in) places where they need to be

### CODE REVIEW

File: schema-guide.md [hunk 1]
```diff
@@ -2315,6 +2315,42 @@ Note that these keys may still not be optimal for, e.g., Icelandic names which d
       type: generic
     ```
 
+#### How to deal with unknown individual authors?
+
+To enable credit for the individuals that have created a work, 
+it is good practice to cite the respective individuals as authors.
+Sometimes you may not be able to determine the person names of the relevant individuals to create
+[`definitions.person`](#definitionsperson) objects for them,
+for example when a software you cite does not provide a `CTATION.cff` file.
```
Comment (sdruskat) [inline schema-guide.md:2324]:
```suggestion
for example when a software you cite does not provide a `CITATION.cff` file.
```

File: schema-guide.md [hunk 2]
```diff
@@ -2315,6 +2315,42 @@ Note that these keys may still not be optimal for, e.g., Icelandic names which d
...
+Then, the next best thing is to refer to those that you could not determine person names for 
+collectively as a "team" or "project" using the title of the software
```
Comment (sdruskat) [inline schema-guide.md:2326]:
```suggestion
collectively as a "team" or "project" using the title of the work
```

File: schema-guide.md [hunk 3]
```diff
@@ -2315,6 +2315,42 @@ Note that these keys may still not be optimal for, e.g., Icelandic names which d
...
+or dataset in a [`definitions.entity`](#definitionsentity) object:
```
Comment (sdruskat) [inline schema-guide.md:2327]:
```suggestion
in a [`definitions.entity`](#definitionsentity) object:
```

File: schema-guide.md [hunk 4]
```diff
@@ -2315,6 +2315,42 @@ Note that these keys may still not be optimal for, e.g., Icelandic names which d
...
+authors:
+  - name: "The Research Software project"
+```
+```yaml
+authors:
+  - family-names: Spaaks
+    given-names: Jurriaan H.
+  - family-names: Druskat
+    given-names: Stephan
+  - name: "The Research Software project"
```
Comment (sdruskat) [inline schema-guide.md:2339]:
```suggestion
  - name: "The Research Software team"
```

**Comment by sdruskat** [comment]
Merging as discussed in a live review call.

**Comment by alex-konovalov** [comment]
@sdruskat Seems to be useful to GAP: see our suggested citation at https://www.gap-system.org/Contacts/cite.html

**Pull request merged**